### PR TITLE
Use from_dict() in UnivariatePolynomial.

### PR DESCRIPTION
**Pull request by CodeMaxx**
Earlier the constructor was called directly.<br/>
This prevents terms with zero coefficient to go into the polynomial.
Also prevents redundant pass of `degree`.
<br/>
Also changed the `from_dict()` function to allow for polynomial with value zero.

@isuruf @irislq @Sumith1896

### CODE REVIEW

File: symengine/polynomial.cpp [hunk 1]
```diff
@@ -413,11 +413,11 @@ RCP<const UnivariatePolynomial> UnivariatePolynomial::from_vec(const RCP<const S
     return make_rcp<const UnivariatePolynomial>(var, degree, std::move(dict));
 }
 
-RCP<const Basic> UnivariatePolynomial::from_dict(const RCP<const Symbol> &var, map_int_Expr &&d)
+RCP<const UnivariatePolynomial> UnivariatePolynomial::from_dict(const RCP<const Symbol> &var, map_int_Expr &&d)
 {
     auto iter = d.begin();
     while (iter != d.end()) {
-        if (Expression(0) == iter->second) {
+        if (Expression(0) == iter->second and Expression(0) != iter->first) {
```
Comment (CodeMaxx) [inline symengine/polynomial.cpp]: @isuruf I have a doubt. This change is required for [this](https://github.com/symengine/symengine/pull/896/files#diff-8b7281251a4cce49b1d83e34bf33e6fbR275) test case to work. Without this `z->is_zero()` returns `false`. But this change is not required for a [similar test case](https://github.com/symengine/symengine/blob/master/symengine/tests/basic/test_polynomial.cpp#L136) in `UnivariateIntPolynomial`. Why so?

Comment (irislq) [inline symengine/polynomial.cpp]: @CodeMaxx  Can you do one thing and change `UnivariatePolynomial::is_zero()` to return just `dict_.empty()`? Maybe that'll help.

**Comment by irislq** [comment]
@CodeMaxx Another thing is that in derivative.cpp, `make_rcp<const UnivariatePolynomial>` is used. I think this needs to be changed to `from_dict()` (like in `UnivariateIntPolynomial`) or this `univariate_polynomial()`

**Comment by CodeMaxx** [comment]
I rather think, use of `from_dict()` should be removed from `UnivariateIntPolynomial` also since the way `diff()` works , the checks done in `from_dict()` are not required. So call to `from_dict()` will be redundant.

**Comment by CodeMaxx** [comment]
@isuruf Ping

**Comment by irislq** [comment]
@isuruf @Sumith1896 Cmake issues in AppVeyor

File: symengine/polynomial.h [hunk 2]
```diff
@@ -204,10 +204,9 @@ RCP<const UnivariatePolynomial> mul_uni_poly(RCP<const UnivariatePolynomial> a,
                                              RCP<const UnivariatePolynomial> b);
 
 inline RCP<const UnivariatePolynomial>
-univariate_polynomial(RCP<const Symbol> i, unsigned int deg,
-                      const map_int_Expr &&dict)
+univariate_polynomial(RCP<const Symbol> i, map_int_Expr &&dict)
```
Comment (irislq) [inline symengine/polynomial.h]: Why did you remove the `const` for `dict`?

**Comment by irislq** [comment]
@CodeMaxx Could you fix the merge conflicts?

File: symengine/polynomial.h (hunk 2 above)
Comment (CodeMaxx) [inline symengine/polynomial.h]: `from_dict()` uses `erase` to remove member of dictionary with coefficients `0` . This will not be possible for a `const` dictionary

**Comment by CodeMaxx** [comment]
Fixed

**Comment by isuruf** [comment]
Binary files have been committed. You need to remove them from history

**Comment by CodeMaxx** [comment]
Done

**Comment by irislq** [comment]
@isuruf Is this ready to be merged yet?

**Comment by isuruf** [comment]
@CodeMaxx, you need to remove the binary files from history since they are very large files

**Comment by CodeMaxx** [comment]
@isuruf I already removed them 2 days ago

**Comment by isuruf** [comment]
Yes, but they are in the git commit history. If we merge this PR, then the size of the repo will be increased by around 5 times

**Comment by CodeMaxx** [comment]
@isuruf removed them from history

**Comment by isuruf** [comment]
Looks good to me. Let's wait until tests pass

**Comment by isuruf** [comment]
@CodeMaxx, thanks for the PR
